test/custom-parser.test.js

```diff
@@ -251,3 +251,142 @@ test('contentTypeParser shouldn\'t support request with undefined "Content-Type"
     })
   })
 })
+
+test('the content type should be a string', t => {
+  t.plan(1)
+  const fastify = Fastify()
+
+  try {
+    fastify.addContentTypeParser(null, () => {})
+    t.fail()
+  } catch (err) {
+    t.is(err.message, 'The content type should be a string')
+  }
+})
+
+test('the content type cannot be an empty string', t => {
+  t.plan(1)
+  const fastify = Fastify()
+
+  try {
+    fastify.addContentTypeParser('', () => {})
+    t.fail()
+  } catch (err) {
+    t.is(err.message, 'The content type cannot be an empty string')
+  }
+})
+
+test('the content type handler should be a function', t => {
+  t.plan(1)
+  const fastify = Fastify()
+
+  try {
+    fastify.addContentTypeParser('aaa', null)
+    t.fail()
+  } catch (err) {
+    t.is(err.message, 'The content type handler should be a function')
+  }
+})
+
+test('catch all content type parser', t => {
+  t.plan(7)
+  const fastify = Fastify()
+
+  fastify.post('/', (req, reply) => {
+    reply.send(req.body)
+  })
+
+  fastify.addContentTypeParser('*', function (req, done) {
+    var data = ''
+    req.on('data', chunk => { data += chunk })
+    req.on('end', () => {
+      done(data)
+    })
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    request({
```

Comment:
Why not`inject`?

Reply:
Because this test has been created before the api `inject` :joy: